lib/plain/entities/resource-type.ts

```diff
@@ -0,0 +1,83 @@
+import type { RawAxiosRequestHeaders } from 'axios'
+
+import type { OptionalDefaults } from '../wrappers/wrap'
+import type {
+  CollectionProp,
+  GetResourceProviderParams,
+  GetResourceTypeParams,
+} from '../../common-types'
+import type { ResourceTypeProps, UpsertResourceTypeProps } from '../../export-types'
+
+export type ResourceTypePlainClientAPI = {
+  /*
+   * Fetch a Resource Type
+   * @param params entity IDs to identify the Resource Type
+   * @returns the Resource Type
+   * @throws if the request fails, or the Resource Type is not found
+   * @example
+   * ```javascript
+   * const resourceType = await client.resourceType.get({
+   *   organizationId: '<organization_id>',
+   *   appDefinitionId: '<app_definition_id>',
+   *   resourceTypeId: '<resource_type_id>',
+   * });
+   * ```
+   */
+  get(params: OptionalDefaults<GetResourceTypeParams>): Promise<ResourceTypeProps>
+
+  /*
+   * Creates or updates a Resource Type
+   * @param params entity IDs to identify the Resource Type
+   * @param rawData the ResourceType
+   * @returns the created or updated Resource Type
+   * @throws if the request fails, the App Definition or Resource Type is not found, or the payload is malformed
```

Comment:
Why does it throw if Resource Type is not found? This also covers the use case when I want to create a new Resource Type.

Reply:
yeah, it should be Resource Provider